### PR TITLE
Fix bug where the Section Headers are not updated correctly

### PR DESCRIPTION
**Pull request by mmroz**
Fixed equality checking for the TableSectionHeaderFooter so that it changes appropriately.

### CODE REVIEW

**Comment by Campbell-Scott** [comment]
Some context on why this is here:

We wanted an `isEqual(_ other: TableHeaderFooterConfigType?) -> Bool` so we could compare two header/footers and re-render a `TableSection` if there were any state changes. If we added this as a requirement to the `TableHeaderFooterConfigType` protocol, without providing a default implementation, it would become a breaking change for any objects conforming to `TableHeaderFooterConfigType` that didn't implement it. The default implementation only exists to prevent that (it should always just evaluate too `true`, if the headers are of the same type, so that we maintain the previous behaviour of not updating at all).

This fix implements the functionality for the out-of-the-box `TableSectionHeaderFooter` that we provide in ftd, but it removes the default conformance for any other custom implementors of `TableHeaderFooterConfigType`; which is a breaking change.

File: FunctionalTableData/TableSectionHeaderFooter.swift [hunk 1]
```diff
@@ -19,13 +19,7 @@ public protocol TableHeaderFooterConfigType: TableItemConfigType {
 	var height: CGFloat { get }
 }
 
-public extension TableHeaderFooterConfigType {
-	func isEqual(_ other: TableHeaderFooterConfigType?) -> Bool {
-		return (other as? Self) != nil
-	}
-}
```
Comment (Caiopia) [inline FunctionalTableData/TableSectionHeaderFooter.swift]: Removing this is potentially a breaking change for protocol adopters who relied on the default implementation. It should be enough to just override the method like you did below, while keeping this one here as the default.

Comment (g-Off) [reply FunctionalTableData/TableSectionHeaderFooter.swift]: Breaking changes are fine, master is currently in a state of broken changes as it works towards its 2.0 release. If you need non-breaking changes then you stick to the tagged release

Comment (Caiopia) [reply FunctionalTableData/TableSectionHeaderFooter.swift]: Oh yeah, that's right. Though I still think we should keep the default implementation.

Comment (g-Off) [reply FunctionalTableData/TableSectionHeaderFooter.swift]: > Oh yeah, that's right. Though I still think we should keep the default implementation.

I... disagree. The default one isn't great and we don't have a default for CellConfigType's isEqual

Comment (Caiopia) [reply FunctionalTableData/TableSectionHeaderFooter.swift]: Fair enough, no strong opinion.

**Pull request merged**